mountpoint-s3/src/fs.rs

```diff
@@ -307,6 +308,10 @@ pub struct S3FilesystemConfig {
     pub allow_delete: bool,
     /// Storage class to be used for new object uploads
     pub storage_class: Option<String>,
+    /// KMS key type for Server side encryption
+    pub kms_key: Option<KmsKeys>,
+    /// KMS key ID for Server side encryption
+    pub key_id: Option<String>
```

Comment:
I think we probably want an optional enum "Encryption"/"EncryptionConfig"/something better named which can have variants. Both variants will have an optional key ID.

```suggestion
    /// Server-side encryption for new S3 objects, if specified
    pub encryption: Option<Encryption>,
```

I see we also have something similar in `mountpoint-s3-client`... I'm less sure what we want to do there, maybe we want to specifically take each as a field like a normal SDK would do. **Let's discuss this as a team.**